ui/selectors/approvals.ts

```diff
@@ -54,6 +54,10 @@ export function getPendingApprovals(state: ApprovalsMetaMaskState) {
   return Object.values(state.metamask.pendingApprovals ?? []);
 }
 
+export function pendingApprovalsSortedSelector(state: ApprovalsMetaMaskState) {
+  return getPendingApprovals(state).sort((a1, a2) => a1.time - a2.time);
+}
```

Comment:
This may be duplicate: https://github.com/MetaMask/metamask-extension/blob/34543337da562145919c7f6c79d4c05e4683ab8b/ui/pages/confirmations/selectors/confirm.ts#L26

Reply:
Does the one in `confirm.ts` not filter to specific types?

Reply:
Exactly, I opted to create a new one that wouldn't filter any pending approvals.